pkg/collector/corechecks/cluster/kubernetes_eventbundle.go

```diff
@@ -0,0 +1,82 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+// +build kubeapiserver
+
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/ericchiang/k8s/api/v1"
+)
+
+type kubernetesEventBundle struct {
+	objUid        string
+	readableKey   string
+	component     string
+	events        []*v1.Event
+	timeStamp     int64
+	lastTimestamp int64
+	countByAction map[string]int
+}
+
+func newKubernetesEventBundler(objUid string, compName string) *kubernetesEventBundle {
+	return &kubernetesEventBundle{
+		objUid:        objUid,
+		events:        []*v1.Event{},
+		component:     compName,
+		countByAction: make(map[string]int),
+	}
+}
+
+func (k *kubernetesEventBundle) addEvent(event *v1.Event) error {
+	if *event.InvolvedObject.Uid != k.objUid {
+		return fmt.Errorf("mismatching Object UIDs: %s != %s", event.InvolvedObject.Uid, k.objUid)
+	}
+	k.events = append(k.events, event)
+	k.countByAction[fmt.Sprintf("**%s**: %s\n", *event.Reason, *event.Message)] += int(*event.Count)
+	k.timeStamp = int64(math.Max(float64(k.timeStamp), float64(*event.Metadata.CreationTimestamp.Seconds)))
```

Comment:
let's keep `k.timeStamp` and `k.lastTimestamp` as float64 to avoid re-casting everytime